Lib/fontTools/ttLib/tables/_n_a_m_e.py

```diff
@@ -147,6 +147,18 @@ def setName(self, string, nameID, platformID, platEncID, langID):
 		else:
 			self.names.append(makeName(string, nameID, platformID, platEncID, langID))
 
+	def removeName(self, nameID, platformID, platEncID, langID):
+		"""Remove a name record identified by 'nameID', 'platformID', 'platEncID'
+                and 'langID'.
+		"""
+		for i, namerecord in enumerate(self.names):
+			if (namerecord.nameID == nameID and
+				namerecord.platformID == platformID and
+				namerecord.platEncID == platEncID and
+				namerecord.langID == langID):
+				    del self.names[i]
+				    break
```

Comment:
how about
```python
self.names = [
    n for n in self.names
    if n.nameID != nameID
    and n.platformID != platformID
    and n.platEncID != platEncID
    and n.langID != langID
]
```

Reply:
This approach will definitely work better for removeNames. I'll update and include both methods.